From f74fa22b669bd771529b3d3be0a96512f81fe92e Mon Sep 17 00:00:00 2001
From: Luan Rafael Carneiro <luan.r@variscite.com>
Date: Wed, 18 Jun 2025 14:11:12 -0300
Subject: [PATCH] eigen: replace Eigen zip download with Git clone to avoid
 checksum instability
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The auto-generated .zip file from GitLab for the specific Eigen commit was
changing its SHA1 hash over time, which breaks integrity verification in
FetchContent/ExternalProject_Add.

This happens because GitLab dynamically generates archive files, and their
binary content can vary due to changes in file order, timestamps, permissions,
or updates to GitLab’s backend infrastructure — even if the source code remains
unchanged.

To ensure reproducible builds and avoid these issues, the .zip download was
replaced with a Git-based FetchContent declaration.

Signed-off-by: Luan Rafael Carneiro <luan.r@variscite.com>
---
 cmake/deps.txt             | 3 +--
 cmake/external/eigen.cmake | 4 ++--
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/cmake/deps.txt b/cmake/deps.txt
index ba9c2bb73..9a29f0ed9 100644
--- a/cmake/deps.txt
+++ b/cmake/deps.txt
@@ -21,7 +21,6 @@ dlpack;https://github.com/dmlc/dlpack/archive/refs/tags/v0.6.zip;4d565dd2e5b3132
 # it contains changes on top of 3.4.0 which are required to fix build issues.
 # Until the 3.4.1 release this is the best option we have.
 # Issue link: https://gitlab.com/libeigen/eigen/-/issues/2744
-eigen;https://gitlab.com/libeigen/eigen/-/archive/e7248b26a1ed53fa030c5c459f7ea095dfd276ac/eigen-e7248b26a1ed53fa030c5c459f7ea095dfd276ac.zip;be8be39fdbc6e60e94fa7870b280707069b5b81a
 flatbuffers;https://github.com/google/flatbuffers/archive/refs/tags/v1.12.0.zip;ba0a75fd12dbef8f6557a74e611b7a3d0c5fe7bf
 fp16;https://github.com/Maratyszcza/FP16/archive/0a92994d729ff76a58f692d3028ca1b64b145d91.zip;b985f6985a05a1c03ff1bb71190f66d8f98a1494
 fxdiv;https://github.com/Maratyszcza/FXdiv/archive/63058eff77e11aa15bf531df5dd34395ec3017c8.zip;a5658f4036402dbca7cebee32be57fb8149811e1
@@ -55,4 +54,4 @@ tensorboard;https://github.com/tensorflow/tensorboard/archive/373eb09e4c5d2b3cc2
 cutlass;https://github.com/NVIDIA/cutlass/archive/refs/tags/v3.1.0.zip;757f90a795034a89d4f48a79d1f009f7a04c8dee
 utf8_range;https://github.com/protocolbuffers/utf8_range/archive/72c943dea2b9240cd09efde15191e144bc7c7d38.zip;9925739c9debc0efa2adcb194d371a35b6a03156
 extensions;https://github.com/microsoft/onnxruntime-extensions/archive/94142d8391c9791ec71c38336436319a2d4ac7a0.zip;4365ac5140338b4cb75a39944a4be276e3829b3c
-composable_kernel;https://github.com/ROCmSoftwarePlatform/composable_kernel/archive/5356c4a943a35e74d7cdc69486afcb8703b9a59a.zip;522382c2af437e09124287e5879ab64af5b2e299
\ No newline at end of file
+composable_kernel;https://github.com/ROCmSoftwarePlatform/composable_kernel/archive/5356c4a943a35e74d7cdc69486afcb8703b9a59a.zip;522382c2af437e09124287e5879ab64af5b2e299
diff --git a/cmake/external/eigen.cmake b/cmake/external/eigen.cmake
index b123adb62..57ef66cb9 100644
--- a/cmake/external/eigen.cmake
+++ b/cmake/external/eigen.cmake
@@ -5,8 +5,8 @@ if (onnxruntime_USE_PREINSTALLED_EIGEN)
 else ()
     FetchContent_Declare(
         eigen
-        URL ${DEP_URL_eigen}
-        URL_HASH SHA1=${DEP_SHA1_eigen}
+        GIT_REPOSITORY https://gitlab.com/libeigen/eigen.git
+        GIT_TAG        e7248b26a1ed53fa030c5c459f7ea095dfd276ac
     )
 
     FetchContent_Populate(eigen)
-- 
2.49.0

